### PR TITLE
Extract room screen action handling into separate component

### PR DESCRIPTION
**Pull request by stefanceriu**
This is hopefuly the first step of many that will simplify the room screen VM, clarify the differences between it and the room timeline controller and make our life easier

### CODE REVIEW

File: ElementX/Sources/Screens/RoomScreen/RoomScreenActionsHandler.swift [hunk 1]
```diff
@@ -0,0 +1,479 @@
...
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+
+import Combine
+import UIKit
+
+enum RoomScreenActionsHandlerAction {
```
Comment (pixlwave) [inline ElementX/Sources/Screens/RoomScreen/RoomScreenActionsHandler.swift:20]: I think the repetition of Action here is enough for me to suggest this file should be something more like `RoomScreenInteractionHandler` or similar?

**Review by pixlwave — APPROVED** [review]

File: ElementX/Sources/Screens/RoomScreen/RoomScreenActionsHandler.swift (hunk 1 above)
Comment (stefanceriu) [reply ElementX/Sources/Screens/RoomScreen/RoomScreenActionsHandler.swift:20]: Yeah, I like that 👍

**Review by nimau — APPROVED** [review]
LGTM 👍

**Review by alfogrillo — APPROVED** [review]
LGTM

**Comment by pixlwave** [comment]
Oh I forgot to mention, should we also revert `swiftlint.yml` back to erroring at 1000 lines again?

**Comment by stefanceriu** [comment]
> should we also revert `swiftlint.yml` back to erroring at 1000 lines again?

yep yep, all done 👍

**Pull request merged**